### PR TITLE
fix WP proxy (WP-636)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

File: inc/Smartling/Helpers/WordpressFunctionProxyHelper.php [hunk 1]
```diff
@@ -64,7 +64,7 @@ public function get_terms()
 
     public function url_to_postid(string $url): int
     {
-        return (int)$this->url_to_postid($url);
+        return (int)url_to_postid($url);
```
Comment (dimitrystd) [inline inc/Smartling/Helpers/WordpressFunctionProxyHelper.php:67]: It looks like method calls itself. Infinite recursion?

File: tests/IntegrationTests/tests/AttachmentsTest.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace IntegrationTests\tests;
+
+use Smartling\Helpers\SiteHelper;
+use Smartling\Helpers\TranslationHelper;
+use Smartling\Submissions\SubmissionManager;
+use Smartling\Tests\IntegrationTests\SmartlingUnitTestCaseAbstract;
+
+class AttachmentsTest extends SmartlingUnitTestCaseAbstract
```
Comment (dimitrystd) [inline tests/IntegrationTests/tests/AttachmentsTest.php:10]: Class name doesn't correlate with test name. Do we really rewrite external links as part of attachments handling?

**Review by dimitrystd — APPROVED** [review]

**Review by dimitrystd — CHANGES_REQUESTED** [review]
Let's talk about how it may help client

File: inc/Smartling/Helpers/WordpressFunctionProxyHelper.php (hunk 1 above)
Comment (PavelLoparev) [reply inc/Smartling/Helpers/WordpressFunctionProxyHelper.php:67]: I believe it was infinite recursion but now it's just wrapper (for tests purposes?) for `url_to_postid` wp function.

**Review by PavelLoparev — APPROVED** [review]

File: tests/IntegrationTests/tests/AttachmentsTest.php (hunk 2 above)
Comment (vsolovei-smartling) [reply tests/IntegrationTests/tests/AttachmentsTest.php:10]: We rewrite links to post based content

File: inc/Smartling/Helpers/WordpressFunctionProxyHelper.php (hunk 1 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Helpers/WordpressFunctionProxyHelper.php:67]: Yes

**Comment by vsolovei-smartling** [comment]
> Let's talk about how it may help client

Without this, code ends up in infinite recursion. Some php configurations are able to catch this condition, most are not.

**Review by dimitrystd — APPROVED** [review]

**Pull request merged**